pymodels/SI_V25_02/lattice.py

```diff
@@ -0,0 +1,865 @@
+    indices = _pyacc_lat.find_indices(the_ring, 'fam_name', 'BC')
+    for i in indices:
+        e = the_ring[i]
+        ang, lng = e.angle, e.length
+        rho = lng/ang
+        if rho < rho0:
+            e.hmin, e.hmax, e.vmin, e.vmax = bc_hfield_vchamber
+    indices = _pyacc_lat.find_indices(the_ring, 'fam_name', 'mc')
+    for i in indices:
+        e = the_ring[i]
+        e.hmin, e.hmax, e.vmin, e.vmax = bc_hfield_vchamber
+        e = the_ring[i + 1]
+        e.hmin, e.hmax, e.vmin, e.vmax = bc_hfield_vchamber
+
+    return the_ring
+
+
+def get_optics_mode(optics_mode=default_optics_mode):
+    """Return magnet strengths for a given optics mode."""
+    mode, version = optics_mode.split('.')
+
+    if mode == 'S05':
+        if version == '01':
+            strengths = {
+                #  QUADRUPOLES
+                #  ===========
+                'Q1': +2.818370601288,
+                'Q2': +4.340329381668,
+                'Q3': +3.218430939674,
+                'Q4': +3.950686823494,
+
+                # NOTE: these values need updating after optics resimetrization with MAD
+                'QDA':  -1.619540412181686,
+                'QFA':  +3.5731777226094446,
+                'QFB':  +4.115082809275146,
+                'QFP':  +4.115082809275146,
+                'QDB1': -2.00677456404202,
+                'QDB2': -3.420574744932221,
+                'QDP1': -2.00677456404202,
+                'QDP2': -3.420574744932221,
+
+                #  SEXTUPOLES
+                #  ===========
+                'SDA0': -80.8337,
+                'SDB0': -64.9422,
+                'SDP0': -64.9422,
+                'SFA0': +52.5696,
+                'SFB0': +73.7401,
+                'SFP0': +73.7401,
+
+                'SDA1': -163.0062328090773,
+                'SDA2': -88.88255991288263,
+                'SDA3': -139.94153649641189,
+                'SFA1': +191.76738248436368,
+                'SFA2': +150.74610044115283,
+                'SDB1': -141.68687364847958,
+                'SDB2': -122.31573949946443,
+                'SDB3': -173.8347917755106,
+                'SFB1': +227.7404567527413,
+                'SFB2': +197.7495405020359,
+                'SDP1': -142.31415019209263,
+                'SDP2': -122.28457189976633,
+                'SDP3': -174.1745194336169,
+                'SFP1': +229.17648360831797,
+                'SFP2': +198.4525009917773,
+            }
+        else:
+            raise _pyacc_acc.AcceleratorException('Version not Implemented')
+    else:
+        raise _pyacc_acc.AcceleratorException('Mode not Implemented.')
+
+    return strengths
+
+
+def create_id_kickmaps_dict(ids):
+    """Return dict with half insertion device kickmaps."""
+    drift = _pyacc_ele.drift
+    
+    if ids:
+        idsdict = {id.subsec:id for id in ids}
+    else:
+        idsdict = dict()
+
+    kickmaps = dict()
+    
+    # CARNAUBA
+    idkey = 'ID06SB'
+    if ids is None or idkey not in idsdict:
```

Comment:
@xresende, you can remove the condition `ids is None` from all if statements.

Reply:
@fernandohds564, this function structure changed from the time you opened your review and when you committed. But the new structure has the same redundancy. I'll clean it.

Reply:
humm, I didn't understand what you said, but if you'll clean the code its ok!